azurerm/internal/services/maintenance/maintenance_assignment_virtual_machine_resource.go

```diff
@@ -0,0 +1,169 @@
+		return fmt.Errorf("creating Maintenance Assignment (virtual machine ID: %q): %+v", virtualMachineIdRaw, err)
+	}
+
+	resp, err := getMaintenanceAssignmentVirtualMachine(ctx, client, virtualMachineId, virtualMachineIdRaw)
+	if err != nil {
+		return err
+	}
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID of Maintenance Assignment (virtual machine ID %q)", virtualMachineIdRaw)
+	}
+
+	d.SetId(*resp.ID)
+	return resourceArmMaintenanceAssignmentVirtualMachineRead(d, meta)
+}
+
+func resourceArmMaintenanceAssignmentVirtualMachineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Maintenance.ConfigurationAssignmentsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.MaintenanceAssignmentVirtualMachineID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	assignment, err := getMaintenanceAssignmentVirtualMachine(ctx, client, id.VirtualMachineId, id.VirtualMachineIdRaw)
+	if err != nil {
+		return err
+	}
+	if assignment.ID == nil || *assignment.ID == "" {
+		return fmt.Errorf("empty or nil ID of Maintenance Assignment (virtual machine ID id: %q", id.VirtualMachineIdRaw)
+	}
+
+	// in list api, `ResourceID` returned is always nil
+	d.Set("virtual_machine_id", id.VirtualMachineIdRaw)
+	if props := assignment.ConfigurationAssignmentProperties; props != nil {
+		d.Set("maintenance_configuration_id", props.MaintenanceConfigurationID)
+	}
+	return nil
+}
+
+func resourceArmMaintenanceAssignmentVirtualMachineDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Maintenance.ConfigurationAssignmentsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.MaintenanceAssignmentVirtualMachineID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.Delete(ctx, id.VirtualMachineId.ResourceGroup, "Microsoft.Compute", "virtualMachines", id.VirtualMachineId.Name, id.Name); err != nil {
+		return fmt.Errorf("deleting Maintenance Assignment to resource %q: %+v", id.VirtualMachineIdRaw, err)
+	}
+
+	return nil
+}
+
+func getMaintenanceAssignmentVirtualMachine(ctx context.Context, client *maintenance.ConfigurationAssignmentsClient, id *parseCompute.VirtualMachineId, virtualMachineId string) (result maintenance.ConfigurationAssignment, err error) {
+	resp, err := client.List(ctx, id.ResourceGroup, "Microsoft.Compute", "virtualMachines", id.Name)
+
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp.Response) {
+			err = fmt.Errorf("checking for presence of existing Maintenance assignment (virtual machine ID: %q): %+v", virtualMachineId, err)
+			return
+		}
+		return result, nil
+	}
+	if resp.Value == nil || len(*resp.Value) == 0 {
+		err = fmt.Errorf("could not find Maintenance assignment (virtual machine ID: %q)", virtualMachineId)
+		return
+	}
+
+	return (*resp.Value)[0], nil
```

Comment:
As above - Can we be sure that the value we need will always be the first (or only) element in this list?

Reply:
above